lib/document.js

```diff
@@ -3703,8 +3703,10 @@ Document.prototype.$getAllSubdocs = function(options) {
   const subDocs = [];
   function getSubdocs(doc) {
     const newSubdocs = [];
-    for (const { path } of doc.$__schema.childSchemas) {
-      const val = doc.$__getValue(path);
+
+    for (const { model } of doc.$__schema.childSchemas) {
+      // Avoid using `childSchemas.path` to avoid compatibility versions with pre-8.8 versions of Mongoose
+      const val = doc.$__getValue(model.path);
```

Comment:
Should this workaround be reverted for the next major version?
If it should, maybe consider adding a TODO or some other kind of reminder.

Reply:
I don't see any reason to prioritize removing this workaround, unless the extra logic in `getValue()` for handling maps causes issues.